tests/test_getPlotDatasetTypeNames.py

```diff
@@ -0,0 +1,113 @@
+from lsst.analysis.tools.tasks.reconstructor import getPlotDatasetTypeNames
+from lsst.daf.butler import Butler, DatasetType
+from lsst.daf.butler.tests.utils import makeTestTempDir, removeTestTempDir
+from lsst.pex.config import Config
+
+TESTDIR = os.path.abspath(os.path.dirname(__file__))
+
+
+class TestGetPlotDatasetTypeNames(TestCase):
+    """Test plot dataset type names are correctly queried by the butler."""
+
+    def setUp(self):
+        self.root = makeTestTempDir(TESTDIR)
+        Butler.makeRepo(self.root)
+        self.butler = Butler(self.root, run="test_run")
+        # No dimensions in dataset type so we don't have to worry about
+        # inserting dimension data or defining data IDs.
+        self.datasetTypePlot = DatasetType(
+            "plotLabel_config",
+            dimensions=(),
+            storageClass="Config",
+            universe=self.butler.registry.dimensions,
+        )
+        self.butler.registry.registerDatasetType(self.datasetTypePlot)
+        self.datasetTypeBase = DatasetType(
+            "baseLabel_config",
+            dimensions=(),
+            storageClass="Config",
+            universe=self.butler.registry.dimensions,
+        )
+        self.butler.registry.registerDatasetType(self.datasetTypeBase)
+        self.datasetTypeOther = DatasetType(
+            "otherLabel_config",
+            dimensions=(),
+            storageClass="Config",
+            universe=self.butler.registry.dimensions,
+        )
+        self.butler.registry.registerDatasetType(self.datasetTypeOther)
+
+    def tearDown(self):
+        removeTestTempDir(self.root)
+
+    def testGetAllPlotNames(self):
+        """Test that we can get all plot dataset type names from a given
+        collection in a butler."""
+        plotConfig = ObjectTableTractAnalysisConfig()
+        plotConfig.plots.skyObjectHistPlot = SkyObjectHistPlot()
+        baseConfig = AnalysisBaseConfig()
+        baseConfig.connections.outputName = "baseName"
+        otherConfig = Config()
+        _ = self.butler.put(plotConfig, "plotLabel_config")
+        _ = self.butler.put(baseConfig, "baseLabel_config")
+        _ = self.butler.put(otherConfig, "otherLabel_config")
+        plots = getPlotDatasetTypeNames(self.butler, "test_run")
+        for plot in plots:
+            self.assertIn("skyObjectHistPlot", plot)
+
+    def testGetLabelPlotNames(self):
+        """Test that we can get plot dataset type names prefixed with a certain
+        label from a given collection in a butler."""
+        plotConfig = ObjectTableTractAnalysisConfig()
+        plotConfig.plots.skyObjectHistPlot = SkyObjectHistPlot()
+        baseConfig = AnalysisBaseConfig()
+        baseConfig.connections.outputName = "baseName"
+        otherConfig = Config()
+        _ = self.butler.put(plotConfig, "plotLabel_config")
+        _ = self.butler.put(baseConfig, "baseLabel_config")
+        _ = self.butler.put(otherConfig, "otherLabel_config")
```

Comment:
I think these other two should also have plot config types, so that you're clearly checking the "did I only get plots with name `plotLabel`" logic.